embulk-jruby-extension/src/main/java/org/jruby/util/RubyDateParser.java

```diff
@@ -0,0 +1,605 @@
+                case FORMAT_OUTPUT:
+                    continue; // skip
+                case FORMAT_STRING:
+                {
+                    String s = token.getData().toString();
+                    for (int si = 0; si < s.length(); ) {
+                        char sc = s.charAt(si);
+                        if (isSpace(sc)) {
+                            while (pos < text.length() && isSpace(text.charAt(pos))) {
+                                pos++;
+                            }
+                        } else if (pos >= text.length() || sc != text.charAt(pos)) {
+                            values.fail();
+                        } else {
+                            pos++;
+                        }
+                        si++;
+                    }
+                    break;
+                }
+                case FORMAT_WEEK_LONG: // %A - The full weekday name (``Sunday'')
+                case FORMAT_WEEK_SHORT: // %a - The abbreviated name (``Sun'')
+                {
+                    int dayIndex = matchAtPatterns(text, pos, dayNames);
+                    if (dayIndex >= 0) {
+                        values.wday = dayIndex % 7;
+                        pos += dayNames[dayIndex].length();
+                    } else {
+                        values.fail();
+                    }
+                    break;
+                }
+                case FORMAT_MONTH_LONG: // %B - The full month name (``January'')
+                case FORMAT_MONTH_SHORT: // %b, %h - The abbreviated month name (``Jan'')
+                {
+                    int monIndex = matchAtPatterns(text, pos, monNames);
+                    if (monIndex >= 0) {
+                        values.mon = monIndex % 12 + 1;
+                        pos += monNames[monIndex].length();
+                    } else {
+                        values.fail();
+                    }
+                    break;
+                }
+                case FORMAT_CENTURY: // %C - year / 100 (round down.  20 in 2009)
+                {
+                    long c;
+                    if (matchAtNumPatterns(nextToken(compiledPattern, i))) {
+                        c = readDigits(2);
+                    } else {
+                        c = readDigitsMax();
+                    }
+                    values._cent = (int)c;
+                    break;
+                }
+                case FORMAT_DAY: // %d, %Od - Day of the month, zero-padded (01..31)
+                case FORMAT_DAY_S: // %e, %Oe - Day of the month, blank-padded ( 1..31)
+                {
+                    long d;
+                    if (text.charAt(pos) == ' ') { // brank or not
+                        pos += 1; // brank
+                        d = readDigits(1);
+                    } else {
+                        d = readDigits(2);
+                    }
+
+                    if (!isValidRange(d, 1, 31)) {
+                        values.fail();
```

Comment:
values need to include some messages why the parsing failed. In CRuby, error message is:

```
irb(main):002:0> Time.strptime('40', '%e')
ArgumentError: invalid strptime format - `%e'
	from /Users/frsyuki/.rbenv/versions/2.2.0/lib/ruby/2.2.0/time.rb:427:in `strptime'
	from (irb):2
	from /Users/frsyuki/.rbenv/versions/2.2.0/bin/irb:11:in `<main>'
```